chapter02/worksheets/solution2.5.2.2.scala

```diff
@@ -0,0 +1,25 @@
+/** 
+  Exercise 2.5.2.2
+  Compute the Collatz sequence 𝑐𝑖 as a stream defined by
+  𝑐₀ = 𝑛 ;
+  𝑐 ₖ+₁ =
+           𝑐ₖ/2         if 𝑐ₖ is even,
+           3 ∗ 𝑐ₖ + 1   if 𝑐ₖ is odd.
+  Stop the stream when it reaches 1 (as one would expect5 it will).
+  */
+
+def collatzSequence(n: Int, xs: List[Int] = List(0)): List[Int] =
+  if xs.head == 1 then xs.init
```

Comment:
You need to use syntax that is compatible with both Scala 2 and Scala 3. This code uses `if then else`, which works only with Scala 3.